server/server.go

```diff
@@ -583,6 +583,19 @@ func (s *Server) startReloadWatcher(ctx context.Context) {
 	log.Infof(ctx, "Registered SIGHUP reload watcher")
 }
 
+// reloadConfigByHUP reload the config and UpdateImageCache
+func (s *Server) reloadConfigByHUP() error {
+	if err := s.config.Reload(); err != nil {
+		return err
+	}
+	if len(s.config.PinnedImages) > 0 {
```

Comment:
I think instead of checking the length, how about we change the logic of `ReloadPinnedImages` to return a boolean type? This will ensure that `UpdateImageCache` is called only when `ReloadPinnedImages` returns true indicating an update to the list of pinned_images.